packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/model/inheritance/single-discriminator/src/Generated/Models/Bird.cs

```diff
@@ -0,0 +1,30 @@
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace _Type.Model.Inheritance.SingleDiscriminator.Models
+{
+    public abstract partial class Bird
+    {
+        protected Bird(string kind, int wingspan) => throw null;
```

Comment:
this is different from our code in autorest.csharp.
Our old code is
```csharp
protectd Bird(int wingspan)
```
and in derived classes we have:
```csharp
public Eagle(int wingspan)
{
Kind = "Eagle";
}
```